app.js

```diff
@@ -141,7 +141,7 @@ document.addEventListener('DOMContentLoaded', function () {
         // First, ensure that the URL is decoded (handles encoded URLs)
         url = decodeURIComponent(url);
     
-        const regex = /^https?:\/\/(www\.youtube\.com\/watch\?v=|youtu\.be\/)([^&?\/\s]+)((\?.*)?)$/;
+        const regex = /^https?:\/\/(www\.youtube\.com\/watch\?v=|youtu\.be\/|music\.youtube\.com\/watch\?v=)(.{11}).*/;
```

Comment:
Good idea, I guess the regex could be made simpler, since only "www." and "music." are the difference in the links

Reply:
Absolutely correct, this was done to not mess up the groups so only need to change this line.